1_Lyalyushkin/AppList.py

```diff
@@ -0,0 +1,85 @@
+CGWindowListCopyWindowInfo.restype = CFArrayRef
+CGWindowListCopyWindowInfo.argtypes = [ctypes.c_uint32, ctypes.c_uint32]
+
+def getStringDictValue(dictionary, key):
+    cfString = CoreFoundation.CFDictionaryGetValue(dictionary, unicode_to_cfstring(key))
+    return cfstring_to_unicode(cfString);
+
+def getIntDictValue(dictionary, key):
+    nsNumber = CoreFoundation.CFDictionaryGetValue(dictionary, unicode_to_cfstring(key))
+    return objc.objc_msgSend(nsNumber, objc.sel_registerName('intValue'))
+
+def getStringProperty(obj, propertyName):
+    result = 'None'
+    nsString = objc.objc_msgSend(obj, objc.sel_registerName(propertyName))
+    if nsString is not None:
+        result = ctypes.string_at(objc.objc_msgSend(nsString, objc.sel_registerName('UTF8String')))
+    return result
+
+def main():
+    NSAutoreleasePool = objc.objc_getClass('NSAutoreleasePool')
+    pool = objc.objc_msgSend(NSAutoreleasePool, objc.sel_registerName('alloc'))
+    pool = objc.objc_msgSend(pool, objc.sel_registerName('init'))
+
+    NSWorkspace = objc.objc_getClass('NSWorkspace')
+    workspace = objc.objc_msgSend(NSWorkspace, objc.sel_registerName('sharedWorkspace'))
+
+    runningApps = objc.objc_msgSend(workspace,objc.sel_registerName('runningApplications'))
+    appCount = objc.objc_msgSend(runningApps, objc.sel_registerName('count'))
+
+    allWindowsList = CGWindowListCopyWindowInfo(0, 0)
+    allWindowsCount = CoreFoundation.CFArrayGetCount(allWindowsList)
+
+    onScreenWindowsList = CGWindowListCopyWindowInfo((1 << 0),0)
+    onScreenWindowsCount = CoreFoundation.CFArrayGetCount(onScreenWindowsList)
+
+    for appIndex in range(appCount):
+        app = objc.objc_msgSend(runningApps, objc.sel_registerName('objectAtIndex:'), appIndex)
+
+        bid = getStringProperty(app, 'bundleIdentifier')
+        pid = objc.objc_msgSend(app, objc.sel_registerName('processIdentifier'))
+
+        print "{}: {}".format(bid, pid)
+
+        for allWindowIndex in range(allWindowsCount):
+            window = CoreFoundation.CFArrayGetValueAtIndex(allWindowsList, allWindowIndex)
+
+            localizedName = getStringProperty(app, 'localizedName')
+            windowOwnerName = getStringDictValue(window, 'kCGWindowOwnerName')
+
+            if windowOwnerName == localizedName:
+                isHidden = True
+                windowNum = 0
+                for onScreenWindowsIndex in range(onScreenWindowsCount):
+                    onScreenWindow = CoreFoundation.CFArrayGetValueAtIndex(onScreenWindowsList, onScreenWindowsIndex)
+
+                    onScreenWindowNum = getIntDictValue(onScreenWindow, 'kCGWindowNumber')
+                    windowNum = getIntDictValue(window, 'kCGWindowNumber')
+
+                    if onScreenWindowNum == windowNum:
+                        isHidden =False
+                        break
+                windowName = getStringDictValue(window, 'kCGWindowName')
+                print "     '{}':{}:{}".format(
```

Comment:
Use `print(...)` function (you need to call `from __future__ import print_function` at the beginning of the module to make it Py2/Py3 compatible).